src/bdc/steps/social_media_api.py

```diff
@@ -0,0 +1,72 @@
+# SPDX-License-Identifier: MIT
+# SPDX-FileCopyrightText: 2023 Ahmed Sheta <ahmed.sheta@fau.de>
+
+from http import HTTPStatus
+
+import facebook
+import pandas as pd
+import requests
+from tqdm import tqdm
+
+from bdc.steps.step import Step
+from config import FACEBOOK_APP_SECRET, FACEBOOK_SDK_GRAPH_API_KEY
+
+
+class FacebookGraphAPI(Step):

```

Comment:
You might also want to consider parent class' log function instead of printing.

Reply:
Addressed